tk: fix x64 darwin build

paparodeo opened the pull request:
update replace -framework -> -weak_framework to remove reference to UniformTypeIdentifiers

https://hydra.nixos.org/build/279564344

staging doesn't need the fix because x64 darwin is on sdk 11
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc
@emilazy

emilazy commented:
This came up in https://github.com/NixOS/nixpkgs/pull/357222 – might be better to just use the macOS 11 SDK for Tk across the board and drop the condition?

paparodeo commented:
> This came up in #357222 – might be better to just use the macOS 11 SDK for Tk across the board and drop the condition?

sure. this change probably breaks tk-8_5 anyway. will remove condition, and build tk-8_5, tk-9 and tk-8_6.

paparodeo commented:
> > This came up in #357222 – might be better to just use the macOS 11 SDK for Tk across the board and drop the condition?
> 
> sure. this change probably breaks tk-8_5 anyway. will remove condition, and build tk-8_5, tk-9 and tk-8_6.

ok. tk-9_0 builds but tcl-8_5 fails due to a compile error so left as broken -- it's trying to typedef ptrdiff_t to an int which seems bad given that it should be 64 bits so there might be other issues.

paparodeo committed:
tk: fix x64 darwin build

update replace -framework -> -weak_framework to remove reference to
UniformTypeIdentifiers

paparodeo commented:
eh. I think moving to apple-sdk_11 is a breaking change for some users on x64 darwin so went with the first option. given that 8.5 and 9.0 are already listed as broken the change doesn't effect them.

vcunat commented:
I think this should be for `staging-next-24.11`.  Right now nix-review-tools-reports says that ~1.4k builds is blocked by this.

paparodeo commented:
> I think this should be for `staging-next-24.11`. Right now nix-review-tools-reports says that ~1.4k builds is blocked by this, and I don't expect this PR rebuilds anything but the broken build.

fixed
